Python/initconfig.c

```diff
@@ -2003,6 +2017,14 @@ config_parse_cmdline(PyConfig *config, PyWideStringList *warnoptions,
             config->use_hash_seed = 0;
             break;
 
+        case 'p':
+            if (wcscmp(_PyOS_optarg, L" new") == 0) {
```

Comment:
Another question that I haven't found the answer to is why we need the leading space in the comparison here. It doesn't seem to work this way for the other command-line options.

Reply:
Are you sure? For me it only works if I remove the leading space.

Perhaps your test is somehow testing it with the single argument `"-p new"` ? There are two forms that should work: a single argument `"-pnew"` or two arguments `"-p" "new"`.

FWIW the only way I've found to verify which parser is being used is to depend on a known bug (#31):
```
(v39) ~/cpygen$ ./python.exe -c 'f(x=1,x=1)'
  File "<string>", line 1
SyntaxError: keyword argument repeated
(v39) ~/cpygen$ ./python.exe -pnew -c 'f(x=1,x=1)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NameError: name 'f' is not defined
(v39) ~/cpygen$ 
```

This was with my fix; without the fix, I'd have to use `"-p new"`.
```suggestion
            if (wcscmp(_PyOS_optarg, L"new") == 0) {
```